NSFS tests fail to create the NSFS interface deployment due to permission issues

On RP, we can see that these tests fail when attempting to fetch a running pod of the deployment:
```
      **get_pods_having_label(
            f"app={deployment_app_label}",
            ocsci_config.ENV_DATA["cluster_namespace"],
        )[0]
    )
E       IndexError: list index out of range
```

After manually reproducing the issue, we can see the reason why no healthy pod was available at the NSFS interface deployment YAML:
```
 - lastTransitionTime: "2024-06-02T10:51:30Z"
    lastUpdateTime: "2024-06-02T10:51:30Z"
    message: 'pods "nsfs-interface-8786548b-" is forbidden: unable to validate against
      any security context constraint: [provider "anyuid": Forbidden: not usable by
      user or serviceaccount, provider restricted-v2: .containers[0].runAsUser: Invalid
      value: 0: must be in the ranges: [1000620000, 1000629999], provider "restricted":
      Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden:
      not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable
      by user or serviceaccount, provider "noobaa-db": Forbidden: not usable by user
      or serviceaccount, provider "noobaa-endpoint": Forbidden: not usable by user
      or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user
      or serviceaccount, provider "machine-api-termination-handler": Forbidden: not
      usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not
      usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable
      by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user
      or serviceaccount, provider "rook-ceph": Forbidden: not usable by user or serviceaccount,
      provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider
      "rook-ceph-csi": Forbidden: not usable by user or serviceaccount, provider "privileged":
      Forbidden: not usable by user or serviceaccount]'
    reason: FailedCreate
    status: "True"
    type: ReplicaFailure
    ```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -156,6 +156,7 @@ from ocs_ci.utility.utils import (
 )
 from ocs_ci.helpers import helpers, dr_helpers
 from ocs_ci.helpers.helpers import (
+    add_scc_policy,
     create_unique_resource_name,
     create_ocs_object_from_kind_and_name,
     setup_pod_directories,
@@ -5631,16 +5632,16 @@ def vault_tenant_sa_setup_factory(request):
 
 
 @pytest.fixture(scope="session")
-def nsfs_interface_session(request):
-    return nsfs_interface_fixture(request)
+def nsfs_interface_session(request, service_account_factory_session):
+    return nsfs_interface_fixture(request, service_account_factory_session)
 
 
 @pytest.fixture(scope="function")
-def nsfs_interface(request):
-    return nsfs_interface_fixture(request)
+def nsfs_interface(request, service_account_factory):
+    return nsfs_interface_fixture(request, service_account_factory)
 
 
-def nsfs_interface_fixture(request):
+def nsfs_interface_fixture(request, service_account_factory):
     created_deployments = []
 
     def nsfs_interface_deployment_factory(pvc_name, pvc_mount_path="/nsfs"):
@@ -5673,6 +5674,13 @@ def nsfs_interface_fixture(request):
         volumes = nsfs_deployment_data["spec"]["template"]["spec"]["volumes"][0]
         volumes["name"] = pvc_name
         volumes["persistentVolumeClaim"]["claimName"] = pvc_name
+
+        # Set a privileged service account for the NSFS deployment
+        sa_acc = service_account_factory()
+        add_scc_policy(sa_acc)
+        nsfs_deployment_data["spec"]["template"]["spec"][
+            "serviceAccountName"
+        ] = sa_acc.name
 
         if ocsci_config.DEPLOYMENT.get("disconnected"):
             update_container_with_mirrored_image(nsfs_deployment_data)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5681,7 +5681,7 @@ def nsfs_interface_fixture(request, service_account_factory):
             namespace=ocsci_config.ENV_DATA["cluster_namespace"],
         )
         sa_acc = service_account_factory(project=proj_obj)
-        add_scc_policy(sa_acc)
+        add_scc_policy(sa_acc, namespace=ocsci_config.ENV_DATA["cluster_namespace"])
         nsfs_deployment_data["spec"]["template"]["spec"][
             "serviceAccountName"
         ] = sa_acc.name

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5676,7 +5676,9 @@ def nsfs_interface_fixture(request, service_account_factory):
         volumes["persistentVolumeClaim"]["claimName"] = pvc_name
 
         # Set a privileged service account for the NSFS deployment
-        sa_acc = service_account_factory()
+        sa_acc = service_account_factory(
+            project=ocsci_config.ENV_DATA["cluster_namespace"]
+        )
         add_scc_policy(sa_acc)
         nsfs_deployment_data["spec"]["template"]["spec"][
             "serviceAccountName"

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5676,9 +5676,11 @@ def nsfs_interface_fixture(request, service_account_factory):
         volumes["persistentVolumeClaim"]["claimName"] = pvc_name
 
         # Set a privileged service account for the NSFS deployment
-        sa_acc = service_account_factory(
-            project=ocsci_config.ENV_DATA["cluster_namespace"]
+        proj_obj = ocp.OCP(
+            kind=constants.NAMESPACE,
+            namespace=ocsci_config.ENV_DATA["cluster_namespace"],
         )
+        sa_acc = service_account_factory(project=proj_obj)
         add_scc_policy(sa_acc)
         nsfs_deployment_data["spec"]["template"]["spec"][
             "serviceAccountName"

